products/llm_observability/frontend/llmObservabilityTraceLogic.ts

```diff
@@ -182,6 +215,48 @@ export const llmObservabilityTraceLogic = kea<llmObservabilityTraceLogicType>([
         setDisplayOption: ({ displayOption }) => {
             localStorage.setItem('llm-observability-display-option', JSON.stringify(displayOption))
         },
+        initializeMessageStates: ({ inputCount, outputCount }) => {
```

Comment:
Overall these state changes could have a bunch of edge cases that my brain cannot figure out while reading this. I would love a test on this. It might force us to extract some of this state calculation into util functions so they can be more easily tested

Reply:
Done!